pkg/api/encryption.go

```diff
@@ -65,104 +49,54 @@ func writeContentToFile(keyData []byte, fileName string) error {
 	return nil
 }
 
-// Encrypt encrypts data with public key
-func Encrypt(pt []byte, pub *rsa.PublicKey) ([]byte, error) {
-	//hash := sha512.New()
-	//ct, err := rsa.EncryptOAEP(hash, rand.Reader, pub, pt, nil)
-	ct, err := rsa.EncryptPKCS1v15(rand.Reader, pub, pt)
-	if err != nil {
-		log.Println(err)
-		return nil, err
+func GenerateKey() (string, error) {
+	salt := make([]byte, 32)
+	if _, err := rand.Read(salt); err != nil {
+		return "", err
 	}
-	return ct, nil
+	s := fmt.Sprintf("%x", salt)
+	return s, nil
 }
 
```

Comment:
Wow - I don't think I've seen it done like this before. A lot of funky calls. Looks cool!